pymc/sampling/forward.py

```diff
@@ -76,6 +76,28 @@
 _log = logging.getLogger(__name__)
 
 
+def get_constant_coords(trace_coords: dict[str, np.ndarray], model: Model) -> set:
+    """Get the set of coords that have remained constant between the trace and model"""
+    constant_coords = set()
+    for dim, coord in trace_coords.items():
+        current_coord = model.coords.get(dim, None)
+        current_length = model.dim_lengths.get(dim, None)
+        if hasattr(current_length, "get_value"):
```

Comment:
I prefer to be more explicit. `if isinstance(current_length, TensorSharedVariable) : ....get_value() elif isinstance(current_length, TensorConstant): ....data`

Reply:
Implemented in https://github.com/pymc-devs/pymc/pull/7381/commits/694660c5b1fb12ebd91ae6e9d323f5433fda7deb